src/hotspot/share/memory/heapShared.cpp

```diff
@@ -701,7 +697,7 @@ void HeapShared::resolve_classes_for_subgraphs(ArchivableStaticFieldInfo fields[
     TempNewSymbol klass_name = SymbolTable::new_symbol(info->klass_name);
     InstanceKlass* k = SystemDictionaryShared::find_builtin_class(klass_name);
     assert(k != NULL && k->is_shared_boot_class(), "sanity");
-    resolve_classes_for_subgraph_of(k, CHECK);
+    resolve_classes_for_subgraph_of(k, THREAD /* exceptions are ignored */);
```

Comment:
They're not ignored though, they're passed to the caller of this function.  Since this function passes TRAPS, the caller should have a CHECK/CATCH or handle the pending exception.

Reply:
The are passed to the callers of this function as THREAD->_pending_exception will have an something in it.

Reply:
Exceptions aren't ignored, they are cleared by the callees, so you don't expect any exceptions, which is CATCH.